cs-json/src/main/java/io/cloudslang/content/json/actions/AddPropertyToObject.java

```diff
@@ -0,0 +1,166 @@
+    /**
+     * Validates the input for this action.
+     *
+     * @param jsonObject       The value of the <b>jsonObject</b> input.
+     * @param newPropertyName  The value of the <b>newPropertyName</b> input.
+     * @param newPropertyValue The value of the <b>newPropertyValue</b> input.
+     * @param returnResult     The map that will be used as return result for this action.
+     * @return True if input is valid, false otherwise.
+     */
+    private boolean validateInput(String jsonObject, String newPropertyName, String newPropertyValue,
+                                  Map<String, String> returnResult) {
+        return validateJsonObject(jsonObject, returnResult);
+    }
+
+    /**
+     * Validates the <b>jsonObject</b> input.
+     *
+     * @param jsonObject   The value of the <b>jsonObject</b> input.
+     * @param returnResult The map that will be used as return result for this action.
+     * @return True if value of <b>jsonObject</b> is valid, false otherwise.
+     */
+    private boolean validateJsonObject(String jsonObject, Map<String, String> returnResult) {
+        if (StringUtilities.isBlank(jsonObject)) {
+            final String exMsg = "Empty JSON string";
+            populateResult(returnResult, new Exception(exMsg));
+            return false;
+        }
+
+        try {
+            JsonNode jsonRoot = objectMapper.readTree(jsonObject);
+            if (jsonRoot == null || !(jsonRoot.isContainerNode() && jsonRoot.isObject())) {
+                throw new Exception();
+            }
+        } catch (Exception exception) {
+            final String exMsg = Constants.InputNames.JSON_OBJECT + " is not a valid JSON Object";
+            populateResult(returnResult, new Exception(exMsg));
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Does the actual logic for this Action
+     *
+     * @param jsonObject       The value of the <b>jsonObject</b> input.
+     * @param newPropertyName  The value of the <b>newPropertyName</b> input.
+     * @param newPropertyValue The value of the <b>newPropertyValue</b> input.
+     * @param returnResult     The map that will be used as return result for this action.
+     */
+    private void doExecute(String jsonObject, String newPropertyName, String newPropertyValue,
+                           Map<String, String> returnResult) {
+        try {
+            JsonNode jsonRoot = objectMapper.readTree(jsonObject);
+            if(!(jsonRoot instanceof ObjectNode)){
+                throw new Exception("The value could not be added!");
+            }
+            ContainerNode jsonResult = ((ObjectNode) jsonRoot).put(newPropertyName, newPropertyValue);
+            if (jsonResult == null) {
+                throw new Exception("The value could not be added!");
+            }
+            populateResult(returnResult, jsonResult.toString(), null);
+        } catch (Exception ex) {
+            populateResult(returnResult, ex);
+        }
+    }
```

Comment:
please rename this method to something more meaningful and move it to a service class